### PR TITLE
Add support for EC-Lab v11.50 (rebased)

### PR DESCRIPTION
**Pull request by chatcannon**
A rebased version of #95 with the CRLF changes undone and the black formatting split to a separate commit.

@JhonFlash3008 can you check to see if this is what you wanted?

### CODE REVIEW

File: galvani/BioLogic.py
Comment (JhonFlash3008) [inline galvani/BioLogic.py:543]: I added the mpr_file.close() here, it seemed to me it was "cleaner" to close the file when finished reading it

**Comment by JhonFlash3008** [comment]
Sorry for the mess with the formatting. In the future I'll make sure to submit in unix formatting. About the Black autoformat, if you don't want I can get rid of it when I work on this project

File: galvani/BioLogic.py
Comment (chatcannon) [reply galvani/BioLogic.py:556]: I would rather avoid that - if the function takes a file object then it should not close it.

File: galvani/BioLogic.py
Comment (chatcannon) [reply galvani/BioLogic.py]: changed

File: galvani/BioLogic.py
Comment (chatcannon) [reply galvani/BioLogic.py]: removed

File: galvani/BioLogic.py
Comment (chatcannon) [reply galvani/BioLogic.py]: Added a comment in the code.

**Comment by chatcannon** [comment]
@ml-evs I included the latest changes from #95 so if you're happy with this then let's merge it.

**Comment by ml-evs** [comment]
@chatcannon could we hold off a day or two more? I've been sent a load of example files now that the ECLab "upgrade" is rolling out in our lab, some of them don't contain the parsed data I would expect and as of right now I haven't been able to debug why yet. Hoping to grab some time later today or tomorrow to try again with the latest state of this PR.

File: galvani/BioLogic.py
Comment (ml-evs) [inline galvani/BioLogic.py:273]: I'm not sure I understand why this had to be renamed, but the knock-on effect is that old MPR files also get passed with this column name (presumably also the same with `<I>/mA` below, but I understand that one more). Is the point that ECLab itself uses the `dq` version of the name? Perhaps it would be good to add this in the release notes as the one "breaking change".

**Review by ml-evs — APPROVED** [review]
Okay, had time to investigate and it was a "me problem", all the tests files I was provided now work well with this PR! No real comments on the code itself; if breaking changes to the MPR format becomes a common occurrence then we might have to restructure along the more automated lines discussed in the issue, but this is good for now.

The only change I had to make downstream to get essentially the same response back from this version of galvani and the last is that the column name for `dQ/mA.h` changes to `dq/mA.h`, even for older MPR files. Perhaps @chatcannon could just verify if this is necessary before merging (I've left a comment at the point where this changed in the code).

Either way, thanks @JhonFlash3008 for making the time to contribute this back, and @chatcannon for guidance!

File: galvani/BioLogic.py
Comment (chatcannon) [reply galvani/BioLogic.py:273]: Given that ID 23 is "dQ/mA.h" maybe this was always correct? I never understood what the difference is between `Q`, `q`, `<q>` etc.

Comment (chatcannon) [reply galvani/BioLogic.py:273]: We should probably also clean up the comments on IDs 8 and 23 below since those lines are now unique.

Comment (ml-evs) [reply galvani/BioLogic.py:273]: > Given that ID 23 is "dQ/mA.h" maybe this was always correct? I never understood what the difference is between `Q`, `q`, `<q>` etc.

Ah right, didn't notice that. I'm not sure of the difference beyond the fact that a file that contains dq used to parse it as dQ but now (correctly I guess) parses it as dq, so we should add the note.

Comment (ml-evs) [reply galvani/BioLogic.py:273]: Just removed the comments; for the sake of everyone's time I think I'll merge this now without awaiting another round of reviews. @chatcannon of course you can have the final say before we release.

Comment (ml-evs) [reply galvani/BioLogic.py:273]: Scratch that, apparently I don't have access to merging PRs either. @chatcannon could you do the honours (and maybe just remove the comments on top of this PR)?

**Pull request merged**
